src/releases/semver/version.rs

```diff
@@ -0,0 +1,140 @@
+use std::fmt::Display;
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+use crate::step::StepError;
+
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub(crate) struct Version {
+    pub(crate) major: u64,
+    pub(crate) minor: u64,
+    pub(crate) patch: u64,
+    pub(crate) pre: Option<Prerelease>,
+}
+
+impl Ord for Version {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.major
+            .cmp(&other.major)
+            .then(self.minor.cmp(&other.minor))
+            .then(self.patch.cmp(&other.patch))
+            .then({
```

Comment:
Probably not a big performance concern, but consider:

```suggestion
            .then_with(|| self.minor.cmp(&other.minor))
            .then_with(|| self.patch.cmp(&other.patch))
            .then_with(|| {
```